### PR TITLE
Add documentation for Deploying to Heroku to README

### PR DESCRIPTION
**Pull request by lpender**
I think that the process of deploying to Heroku has created enough [confusion](https://github.com/thoughtbot/suspenders/issues/935) and [frustration](https://github.com/thoughtbot/suspenders/issues/919#issuecomment-421632156) that it's time to add a simple summary of the setup required to the documentation about it. This should be updated and expanded or contracted when new issues arise or are resolved.

### CODE REVIEW

**Comment by StevenXL** [comment]
@lpender thank you for submitting this PR. I hope it gets merged soon. It was very helpful for me - i found it a little late but my build kept failing as I had the `APPLICATION_HOST` set incorrectly.

File: README.md [hunk 1]
```diff
@@ -117,6 +117,8 @@ Suspenders also comes with:
 
 ## Heroku
 
+Read the documentation on [deploying to Heroku](heroku deploy)
```
Comment (composerinteralia) [inline README.md]: Broken Link:

```suggestion
Read the documentation on [deploying to Heroku][heroku deploy]
```

Comment (composerinteralia) [reply README.md]: Now I am getting a 404 for the link

File: README.md [hunk 2]
```diff
@@ -131,6 +133,7 @@ This:
 * Schedules automated backups for 10AM UTC for both `staging` and `production`
 
 [Heroku Pipeline]: https://devcenter.heroku.com/articles/pipelines
+[heroku deploy]: https://github.com/thoughtbot/suspenders/docs/heroku_deploy.md
```
Comment (lpender) [inline README.md]:
```suggestion
[heroku deploy]: https://github.com/thoughtbot/suspenders/blob/master/docs/heroku_deploy.md
```

File: README.md (hunk 1 above)
Comment (lpender) [reply README.md]: Hey, apologies. I think there are two issues:

1. The link was actually wrong (didn't contain `/blob/master/`
2. The link won't work until this is actually merged to `master`

Comment (composerinteralia) [reply README.md]: Thanks!

**Pull request merged**